tests/compiler/test_dict.py

```diff
@@ -428,3 +428,56 @@ def lookup(str):
 
     for var, exp_val in zip(vars_to_lookup, expected_values):
         assert lookup(var) == exp_val
+
+
+@pytest.mark.parametrize(
+    "lookup_path, expectation",
+    [
+        ('a["A"]["1"]', True),
+        ('a["A"]["2"]', False),
+        ('a["A"]["3"]', False),
+        ('a["A"]["4"]', False),
+        ('a["A"]', True),
+        ('a["B"]', False),
+        ('a["C"]', False),
+        ('a["D"]', True),
+        ('a["K"]', False),
```

Comment:
```suggestion
        ('a["E"]', True),
        ('a["K"]', False),
```
Tied to the comment below